### PR TITLE
chore: drop vestigial inbox-driver switching by origin

### PR DESCRIPTION
**Pull request by wegry**

### CODE REVIEW

File: src/platform-implementation-js/platform-implementation.js
Comment (wegry) [inline src/platform-implementation-js/platform-implementation.js:1]: Is the `Driver` v. `GmailDriver` distinction something we want to keep around if we make a change like this?

File: src/platform-implementation-js/platform-implementation.js [hunk 1]
```diff
@@ -240,28 +239,17 @@ export function makePlatformImplementation(
...
 
   const origin: string =
     (process.env.NODE_ENV === 'test' && global.__test_origin) ||
     document.location.origin;
-  const DriverClass = DRIVERS_BY_ORIGIN[origin];
-  if (!DriverClass) {
-    console.log('InboxSDK: Unsupported origin', origin);
-    // never resolve
-    return new Promise((resolve, reject) => {});
-  }
```
Comment (Macil) [inline src/platform-implementation-js/platform-implementation.js:260]: We need to keep the logic that if we're on an unrecognized domain we don't initialize the driver. It's common enough for people to import the InboxSDK in a content script that gets run on other domains. We could axe the console.log though: that was present to discourage people from remote-loading us into non-Gmail domains and giving us a large permission we didn't need, but we don't remote-load in the now recommended npm build so it doesn't matter much now.

File: src/platform-implementation-js/platform-implementation.js
Comment (Macil) [reply src/platform-implementation-js/platform-implementation.js:1]: Good question. Let's say no, so we're not keeping an unnecessary abstraction that might not pay off.

File: src/platform-implementation-js/platform-implementation.js (hunk 1 above)
Comment (wegry) [reply src/platform-implementation-js/platform-implementation.js]: [b81ce52](https://github.com/InboxSDK/InboxSDK/pull/875/commits/b81ce52b738292abcda8e30f62783af49e68e199)

File: src/platform-implementation-js/platform-implementation.js
Comment (wegry) [reply src/platform-implementation-js/platform-implementation.js:1]: Dropped in [b81ce52](https://github.com/InboxSDK/InboxSDK/pull/875/commits/b81ce52b738292abcda8e30f62783af49e68e199).

File: src/platform-implementation-js/platform-implementation.js (hunk 1 above)
Comment (wegry) [reply src/platform-implementation-js/platform-implementation.js]: > We need to keep the logic that if we're on an unrecognized domain we don't initialize the driver.

Keeping this logic has the added benefit of patching up #873.

**Pull request merged**